tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CheckAllTool/ChangelogValidationTool.cs

```diff
@@ -0,0 +1,208 @@
+                
+                if (!Directory.Exists(packagePath))
+                {
+                    SetFailure(1);
+                    return new FailureCLICheckResponse(1, "", $"Package path does not exist: {packagePath}");
+                }
+
+                // Find the SDK repository root by looking for common repository indicators
+                // Start from the package path and work upwards to find the SDK repo root
+                var packageRepoRoot = gitHelper.DiscoverRepoRoot(packagePath);
+                if (string.IsNullOrEmpty(packageRepoRoot))
+                {
+                    SetFailure(1);
+                    return new FailureCLICheckResponse(1, "", $"Could not find repository root from package path: {packagePath}");
+                }
+
+                // Construct the path to the PowerShell script in the SDK repository
+                // The script should be in the package's repository root, not relative to this tool's location
+                var scriptPath = Path.Combine(packageRepoRoot, "eng", "common", "scripts", "Verify-ChangeLog.ps1");
+                
+                if (!File.Exists(scriptPath))
+                {
+                    SetFailure(1);
+                    return new FailureCLICheckResponse(1, "", $"PowerShell script not found at expected location: {scriptPath}");
+                }
+
+                // Execute the PowerShell script
+                var result = await ExecuteChangelogValidationScript(scriptPath, packagePath);
+                stopwatch.Stop();
+
+                if (result.Success)
+                {
+                    return new SuccessCLICheckResponse(0, System.Text.Json.JsonSerializer.Serialize(new
+                    {
+                        Message = "Changelog validation completed successfully",
+                        Duration = (int)stopwatch.ElapsedMilliseconds
+                    }));
+                }
+                else
+                {
+                    SetFailure(1);
+                    return new FailureCLICheckResponse(1, "", result.ErrorMessage);
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Unhandled exception while running changelog validation");
+                stopwatch.Stop();
+                SetFailure(1);
+                return new FailureCLICheckResponse(1, "", $"Unhandled exception: {ex.Message}");
+            }
+        }
+
+        /// <summary>
+        /// Executes the PowerShell changelog validation script
+        /// </summary>
+        /// <param name="scriptPath">Path to the PowerShell script</param>
+        /// <param name="packagePath">Path to the package directory</param>
+        /// <returns>Validation result</returns>
+        private async Task<(bool Success, string ErrorMessage, List<string> Details)> ExecuteChangelogValidationScript(string scriptPath, string packagePath)
```

Comment:
This method can also use the process helper now and return a typed response instead of a tuple.

Reply:
One another option is to move these checks into LanguageRepoService and make it as abstract class with implementation for common methods in language repo.

Tools can take language parameter, identify language service and call the implementation.